interface/HemisphereMixer.h

```diff
@@ -268,9 +268,9 @@ template <class EventClass> class HemisphereMixer : public BaseOperator<EventCla
         }
 
         std::vector<std::size_t> index_nns(knn_);
-        std::vector<double> dist_nns(knn_);
+        //std::vector<double> dist_nns(knn_);
```

Comment:
no, that will not work because knn_ does not exist at that point and indeed like that you are creating a a vector of 5 vectors not a vector of vector of 5 elements, use emplace_back as I suggest below